openmc/filter.py

```diff
@@ -728,7 +728,28 @@ class SurfaceFilter(WithIDFilter):
         The number of filter bins
 
     """
-    expected_type = Surface
+
+
+    def __init__(self, bins, filter_id=None):
+        if(type(bins)==list or isinstance(bins, np.ndarray) or 
+           isinstance(bins, openmc.Surface)):
+            bins = np.atleast_1d(bins)
+        else:
+            bins = np.atleast_1d(bins.component_surfaces)
```

Comment:
I think ideally a user would be able to provide the `CompositeSurface` object in the `bins` argument alongside the other surfaces:

```python
surfaces = [openmc.ZCylinder(r=1), openmc.ZCylinder(r=2), openmc.ZCylinder(r=3)]
prism = openmc.model.RectgangularPrism(7, 7)

surface_filter = openmc.SurfaceFilter(surfaces + [prism])
```

Reply:
We'll probably also want to show a warning explaining that many bins will be added for the `CompositeSurface` if one exists to the user isn't confused about extra bins appearing in the tally results when applying the resulting filter.

Reply:
The feature to combine `Surface` and `CompositeSurface` is now available. And I have also added a user warning.